fragmenter/fragment.py

```diff
@@ -766,122 +806,128 @@ def _prepare_molecule(
         return molecule, stereo, found_functional_groups, found_ring_systems
 
     @abc.abstractmethod
-    def fragment(self, **kwargs):
-        """Fragment molecules."""
-        raise NotImplementedError()
+    def fragment(self, molecule: Molecule) -> FragmentationResult:
+        """Fragments a molecule according to this classes settings.
 
-    def get_provenance(self) -> Dict[str, Any]:
-        """
-        Get version of fragmenter and options used
+        Parameters
+        ----------
+        molecule
+            The molecule to fragment.
 
+        Returns
+        -------
+            The results of the fragmentation including the fragments and provenance
+            about the fragmentation.
         """
-        # import getpass
-        # import socket
-        # import uuid
-        #
-        # import fragmenter
-        #
-        # fragmenter_version = fragmenter.__version__
+
+        raise NotImplementedError()
+
+    def _default_provenance(self) -> Dict[str, Any]:
```

Comment:
Will there be other methods of getting provenance, therefore requiring this one to specify default?

Reply:
I think the subclasses may in future need to store extra provenance (e.g. the WBO values for the rotors for the WBOFragmenter), so I would consider this the 'default that must be included'.